### PR TITLE
Fix code logic bug to parse log level

### PR DESCRIPTION
**Pull request by disa6302**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: canary/producer-c/canary/KvsProducerSampleCloudwatch.cpp [hunk 1]
```diff
@@ -260,7 +260,10 @@ INT32 main(INT32 argc, CHAR* argv[])
         }
 
         // adjust members of pDeviceInfo here if needed
-        pDeviceInfo->clientInfo.loggerLogLevel = LOG_LEVEL_DEBUG;
+        logLevel = getenv(DEBUG_LOG_LEVEL_ENV_VAR);
+        if (logLevel != NULL) {
+            STRTOUI32(logLevel, NULL, 10, &pDeviceInfo->clientInfo.loggerLogLevel);
```
Comment (MushMal) [inline canary/producer-c/canary/KvsProducerSampleCloudwatch.cpp:265]: Need to assume some default value if this fails. Something like

if (STATUS_FAILED(STRTOUI(*)) {
    pDeviceInfo->clientInfo.loggerLogLevel = DEFAULT_BLAH_BLAH_LOG_LEVEL;
}

Comment (disa6302) [reply canary/producer-c/canary/KvsProducerSampleCloudwatch.cpp:265]: Default is internally set to WARN.

Comment (disa6302) [reply canary/producer-c/canary/KvsProducerSampleCloudwatch.cpp:265]: Actually, now that I think about it, I can set it to LOG_LEVEL_DEBUG by default and set to provided value if env is set.

Comment (MushMal) [reply canary/producer-c/canary/KvsProducerSampleCloudwatch.cpp:265]: sounds good. I am going to approve it

**Review by MushMal — APPROVED** [review]

**Review by MushMal — APPROVED** [review]

**Pull request merged**